app/logical/sources/strategies/null.rb

```diff
@@ -24,6 +24,122 @@ def normalizable_for_artist_finder?
       def normalize_for_artist_finder
         url
       end
+
+      def normalize_for_source
+        case url
+        when %r{\Ahttps?://(?:d3j5vwomefv46c|dn3pm25xmtlyu)\.cloudfront\.net/photos/large/(\d+)\.}i
+          base_10_id = $1.to_i
+          base_36_id = base_10_id.to_s(36)
+          "https://twitpic.com/#{base_36_id}"
+
+        when %r{\Ahttp://www\.karabako\.net/images(?:ub)?/karabako_(\d+)(?:_\d+)?\.}i
+          "http://www.karabako.net/post/view/#{$1}"
+
+        # XXX http://twipple.jp is defunct
+        # http://p.twpl.jp/show/orig/myRVs
+        when %r{\Ahttp://p\.twpl\.jp/show/(?:large|orig)/([a-z0-9]+)}i
+          "http://p.twipple.jp/#{$1}"
+
+        when %r{\Ahttps?://blog(?:(?:-imgs-)?\d*(?:-origin)?)?\.fc2\.com/(?:(?:[^/]/){3}|(?:[^/]/))([^/]+)/(?:file/)?([^\.]+\.[^\?]+)}i
+          username = $1
```

Comment:
I guess it was already this way before but I'd personally prefer named groups instead of first only numbering them and then naming them each the next line, though it would make the regexes longer

Reply:
This depends on whether @evazion wants me to write multiline regexes. I didn't see them anywhere in the code so I avoided writing them, though it would look cleaner yeah.